### PR TITLE
RA 2100 SFA.DAS.VacanyServices.Wage package

### PR DESCRIPTION
**Pull request by CarlHA**
A package have a common location for National Minimum Wage data plus helper methods to display/ calculate wage information.

### CODE REVIEW

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/WagePresenter.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        public static string GetYearlyApprenticeshipMinimumWage(decimal hoursPerWeek, DateTime? possibleStartDate)
+        {
+            NationalMinimumWageRates wageRange = NationalMinimumWageService.GetHourlyRates(possibleStartDate);
+
+            return
+                $"{FormatWageAmount(WageCalculator.GetYearlyRateFromHourlyRate(wageRange.ApprenticeMinimumWage, hoursPerWeek))}";
+        }
+
+        private static string GetDisplayAmount(WageType type, WageUnit unit, decimal? amount, decimal? amountLowerBound,
```
Comment (chrisjensenuk) [inline VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/WagePresenter.cs]: As discussed. Make this public. Pass in an object with a WageType property. Remove the other helper functions.

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage.UnitTests/WagePresenterTests.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    [Parallelizable]
+    public class WagePresenterTests
+    {
+        private const string Space = " ";
+
+        [TestCase(WageUnit.Weekly, WageConstants.WeeklyWageText)]
+        [TestCase(WageUnit.Annually, WageConstants.AnnualWageText)]
+        [TestCase(WageUnit.Monthly, WageConstants.MonthlyWageText)]
+        [TestCase(WageUnit.NotApplicable, WageConstants.WageText)]
+        public void ShouldGetHeaderDisplayTextForCustomWageType(WageUnit wageUnit, string expected)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage.UnitTests/WagePresenterTests.cs]: Removed

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            Under18NationalMinimumWage = under18NationalMinimumWage;
+            Between18And20NationalMinimumWage = between18And20NationalMinimumWage;
+            Between21And24NationalMinimumWage = between21And24NationalMinimumWage;
+            Over24NationalMinimumWage = over24NationalMinimumWage;
+        }
+        public DateTime ValidFrom { get;  }
+        public DateTime ValidTo { get;  }
+        public decimal ApprenticeMinimumWage { get; }
+        public decimal Under18NationalMinimumWage { get; }
+        public decimal Between18And20NationalMinimumWage { get; }
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs]: Changed

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        public decimal Between21And24NationalMinimumWage { get; }
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs]: Changed

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        public decimal Over24NationalMinimumWage { get; }
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageRates.cs]: Changed

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        private const int WeeksPerYear = 52;
+        private static readonly NationalMinimumWageRates[] nationalMinimumWageRates;
+
+        static NationalMinimumWageService()
+        {
+            nationalMinimumWageRates = GetNationalMinimumWageRates();
+            CurrentRates = GetHourlyRates(DateTime.Today);
+        }
+
+        public static NationalMinimumWageRates CurrentRates { get; }
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Have removed it as it is not used

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+                    weeklyRates.ValidFrom,
+                    weeklyRates.ApprenticeMinimumWage * WeeksPerYear,
+                    weeklyRates.Under18NationalMinimumWage * WeeksPerYear,
+                    weeklyRates.Between18And20NationalMinimumWage * WeeksPerYear,
+                    weeklyRates.Between21And24NationalMinimumWage * WeeksPerYear,
+                    weeklyRates.Over24NationalMinimumWage * WeeksPerYear
+                );
+            }
+
+            NationalMinimumWageRates latestRates = GetWeeklyRates(DateTime.Today, hoursPerWeek);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Done

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        /// <summary>
+        ///     Gets the weekly rates as at a specified date and hours per week.  If no date specified then will return the
+        ///     latest rates. If no hours specified then will use 37.5 hours as a standard week.
+        /// </summary>
+        /// <param name="dateTime">The specified date for the rates</param>
+        /// <param name="hoursPerWeek">The specified hours per week</param>
+        /// <returns>The rates as at the specified date or the latest rates if no date specified</returns>
+        public static NationalMinimumWageRates GetWeeklyRates(DateTime? dateTime, decimal hoursPerWeek = 37.5m)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Done

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        /// <summary>
+        ///     Gets the annual rates as at a specified date and hours per week.  If no date specified then will return the
+        ///     latest rates. If no hours specified then will use 37.5 hours as a standard week.
+        /// </summary>
+        /// <param name="dateTime">The specified date for the rates</param>
+        /// <param name="hoursPerWeek">The specified hours per week</param>
+        /// <returns>The rates as at the specified date or the latest rates if no date specified</returns>
+        public static NationalMinimumWageRates GetAnnualRates(DateTime? dateTime, decimal hoursPerWeek = 37.5m)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Done

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+                    hourlyRates.ValidFrom,
+                    hourlyRates.ApprenticeMinimumWage * hoursPerWeek,
+                    hourlyRates.Under18NationalMinimumWage * hoursPerWeek,
+                    hourlyRates.Between18And20NationalMinimumWage * hoursPerWeek,
+                    hourlyRates.Between21And24NationalMinimumWage * hoursPerWeek,
+                    hourlyRates.Over24NationalMinimumWage * hoursPerWeek
+                );
+            }
+
+            NationalMinimumWageRates latestRates = nationalMinimumWageRates.Single(r => DateTime.Today >= r.ValidFrom && DateTime.Today < r.ValidTo);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Done

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+        public static NationalMinimumWageRates CurrentRates { get; }
+
+        /// <summary>Gets the hourly rates as at a specified date.  If no date specified then will return the latest rates.</summary>
+        /// <param name="dateTime">The specified date for the rates</param>
+        /// <returns>The rates as at the specified date or the latest rates if no date specified</returns>
+        public static NationalMinimumWageRates GetHourlyRates(DateTime? dateTime)
+        {
+            if (dateTime.HasValue)
+                return nationalMinimumWageRates.Single(r => dateTime >= r.ValidFrom && dateTime < r.ValidTo);
+            return nationalMinimumWageRates.Single(r => DateTime.Today >= r.ValidFrom && DateTime.Today < r.ValidTo);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/NationalMinimumWageService.cs]: Done

File: VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/WagePresenter.cs (hunk 1 above)
Comment (CarlHA) [reply VacancyServices.Wage/SFA.DAS.VacancyServices.Wage/WagePresenter.cs]: Now updated

**Pull request merged**